### PR TITLE
Add type to plugin configuration

### PR DESCRIPTION
**Pull request by matissepage**
warning Depends on #2163 

**This PR is still in progress**

## What does this PR do ?

Split Plugin config into another file for more clarity

### How should this be manually tested?

```npm run test:unit```
or
```npm run test```

### CODE REVIEW

File: lib/types/config/PluginsConfiguration.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
+import { JSONObject } from '../../../index';
+
+import { PasswordPolicy } from '../index';
+
+export type PluginsConfiguration = {
+    /**
+     * Common configuration for all plugins.
+     */
+     common: {
```
Comment (Aschen) [inline lib/types/config/PluginsConfiguration.ts:9]: Wrong identation

**Review by Aschen — CHANGES_REQUESTED** [review]
Can you split both plugins config in separate files as well for much clarity?

**Review by Aschen — APPROVED** [review]

**Review by alexandrebouthinon — APPROVED** [review]

**Review by xbill82 — APPROVED** [review]

**Comment by Aschen** [comment]
Can be merged once the conflicts are resolved :+1:

File: lib/types/config/PluginsConfiguration.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    encryption: string;
+
+    /**
+     * If true, Kuzzle will refuse any credentials update or deletion,
+     * unless the currently valid password is provided
+     * or if the change is performed via the security controller.
+     *
+     * @default false
+     */
+    requirePassword: false;
```
Comment (Aschen) [inline lib/types/config/PluginsConfiguration.ts:163]: Is it always `false` or a `boolean`

**Review by Aschen — CHANGES_REQUESTED** [review]

**Pull request merged**